lib/player.js

```diff
@@ -1795,8 +1795,17 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
 
     this.drmEngine_.configure(this.config_.drm);
 
+    const tracksChanged = shaka.util.StreamUtils.applyRestrictions(
+        this.manifest_.variants, this.config_.restrictions, this.maxHwRes_);
+    if (tracksChanged && this.streamingEngine_) {
+      this.onTracksChanged_();
+    }
+
+    const playableVariants = shaka.util.StreamUtils.getPlayableVariants(
```

Comment:
This removes any variant that is not `allowedByKeySystem`.  In some cases, this may prevent license requests for streams that are only unplayable **because** we haven't made a license request.

Reply:
That's correct, but at this point no variant `allowedByKeySystem = false` has yet been set, so it will never filter `allowedByKeySystem`